cypress/page-objects/shopPage.js

```diff
@@ -0,0 +1,34 @@
+import BasePage from './basePage';
+import Chance from 'Chance';
+
+const chance = new Chance();
+
+class ShopPage extends BasePage {
+  selectCategory(allCategories) {
+    return chance.pickone(allCategories.collections);
+  }
+
+  selectProduct(selectedCategory) {
+    const selectedItem = chance.pickone(selectedCategory.items);
+    selectedItem.linkUrl = selectedCategory.linkUrl;
+    return selectedItem;
+  }
+
+  pickRandomProduct(listOfProducts) {
+    const selectedCategory = this.selectCategory(listOfProducts);
+    return this.selectProduct(selectedCategory);
+  }
+
+  addProductToCart(pickedItem, quantity = 1) {
+    this.open(pickedItem.linkUrl);
+    cy.get(`div[name="${pickedItem.name}"]`).parent().find('button').invoke('show')
+      .contains('Add to Cart')
+      .then(($CTA) => {
```

Comment:
`$CTA`
`$` is not required here, it is just a wildcard showing that something is variable, you can remove it

Reply:
gotcha, done